plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -2029,9 +2029,26 @@ def move_gear(gear, destination_container, destination_district_uuid, change_dis
           end
         end
       else
-        # leave quota the same if node profile does not change
-        quota_blocks = Integer(gear_quota[3])
-        quota_files = Integer(gear_quota[6])
+        # if gear is over disk quota limit, increase destination quota by 1% so gear move, start, stop, restart complete
+        # get the base quota blocks and files destination container
+        destination_quota = destination_container.rpc_get_facts_direct(["quota_blocks", "quota_files"])
+        # set destination container quotas
+        quota_blocks = Integer(destination_quota[:quota_blocks])
+        quota_files = Integer(destination_quota[:quota_files])
+        # get current block and file usage from the source container
+        blocks_used = Integer(gear_quota[1])
+        if blocks_used >= quota_blocks
+          if blocks_used > quota_blocks * 1.01
+            log_debug "Cannot move #{gear.group_instance.gear_size} size gear to #{destination_node_profile}.  Gear exceeds quota limits and quota limit buffer also used up."
+            raise OpenShift::UserException.new("Error moving gear. Cannot move gear to #{destination_node_profile} node profile while gear exceeds profile quota limits.")
+          end
+          log_debug "WARNING: Blocks_used exceeds target disk quota, now using buffer"
+          quota_blocks = blocks_used * 1.01
```

Comment:
>@ironcladlou the max amt the quota limit can be increased is
1.01 * Integer(destination_quota[:quota_blocks])

>The destination_quota is always based on the default (1048576 blockmax, 8000 inode max for small gears. The limit will be raised to max of 1059062 then will error rather than give warning.

That might be my key misunderstanding. When `quota_blocks` is reassigned and then fed to sync, the effective quota for this attempt is NOT persisted or otherwise re-associated with that gear. Thanks, and nevermind!